.github/workflows/ci.yml

```diff
@@ -3,6 +3,7 @@ name: Main CI
 on:
   workflow_dispatch:
   push:
+    branches: ['main']
```

Comment:
So the main reason I had done `push` orginally instead of `pull_request` is that I quite enjoy being able to get CI to run without creating a PR -- such as for iteration, etc. It's not _absolutely necessary_, but was a nice thing to have.

I had to go and add `worfklow_dispatch` (for `pull_request`) because the CI workflows would not run on _external-forked_ PRs for `push` events, but would run for `workflow_dispatch` (specifically, `pull_request` events, but with the benefit of also being manually trigger-able).

I suppose since we're the only two mainly contributing to this repo it's less of an issue now, but it was a bigger issue when you had been using a forked approach 🤔 

I did notice that it was running these twice in some cases and was quite annoying, but from what I had found on GitHub's docs, they were "expected" and there was no great workaround. Unless I was mistaken about this 😕 

Thoughts?

Reply:
Hmm, that's quite interesting. I'd like this to be clean and only run once per pull request, and it also saves you some of the minutes you have as a free user for actions. There's a [Skip Duplicate Actions](https://github.com/marketplace/actions/skip-duplicate-actions) action that could be worth a look, but we could probably also build something out of the `if` conditions in jobs. I'll leave the decision to you since it's actions on your account; this was just to avoid all the duplicate runs that this PR will probably generate over its lifespan.

Reply:
That's fair -- to be honest for now it's a non-issue so this is probably fine to keep in here, and we can revisit this if we start actually getting other contributors and notice that some of the CI tasks (specifically the linter) is not getting run.